### PR TITLE
(PC-14060) test(apiHelpers): retro test

### PR DESCRIPTION
**Pull request by bruno-ebstein-pass-culture**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-14060

Une première PR pour ajouter des tests et faire des refactor

## Checklist

I have:

- [x] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [x] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [x] Written **unit tests** native (and web when implementation is different) for my feature.

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/libs/react-native-device-info/__mocks__/getUniqueId.web.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const getUniqueId = jest.fn().mockReturnValue('ad7b7b5a169641e27cadbdb35adad9c4ca23099a')
```
Comment (voisinhugo) [inline src/libs/react-native-device-info/__mocks__/getUniqueId.web.ts:1]: We don't need `src/libs/react-native-device-info/__mocks__/getUniqueId.web.ts` if we have the exact same `src/libs/react-native-device-info/__mocks__/getUniqueId.ts`, do we?

File: src/features/auth/AuthContext.tsx [hunk 2]
```diff
@@ -49,7 +49,8 @@ export const AuthWrapper = memo(function AuthWrapper({ children }: { children: J
       if (getAccessTokenStatus(accessToken) === 'expired') {
         // refreshAccessToken calls the backend to get a new access token
         // and also saves it to the storage
-        accessToken = await refreshAccessToken(api)
+        const { result } = await refreshAccessToken(api)
+        accessToken = result
```
Comment (voisinhugo) [inline src/features/auth/AuthContext.tsx]: Les 2 lignes peuvent être simplifiées par :
`{ result: accessToken } = await refreshAccessToken(api)`

Comment (voisinhugo) [reply src/features/auth/AuthContext.tsx]: Caduque par la suite, ok pour moi

File: src/libs/react-native-device-info/__mocks__/getUniqueId.web.ts (hunk 1 above)
Comment (bruno-ebstein-pass-culture) [reply src/libs/react-native-device-info/__mocks__/getUniqueId.web.ts:1]: `yarn test:unit:web 'apiHelper`

```
 FAIL  src/api/__tests__/apiHelpers.test.ts (5.135 s)
  [api] helpers
    [method] safeFetch
      ✕ should call fetch with populated header (154 ms)
      ✕ should call fetch with populated header when route is in NotAuthenticatedCalls (52 ms)
      ✓ needs authentication response when refresh token fails (39 ms)
      ✕ regenerates the access token and fetch the real url after when the access token is expired (9 ms)
      ✓ needs authentication response when there is no access token (2 ms)
      ✓ needs authentication response when there is no refresh token (1 ms)
```

c'est malheureusement nécessaire, sinon les tests webs appellent la vraie implem

**Review by voisinhugo — APPROVED** [review]

**Pull request merged**